src/main/java/org/spongepowered/mod/mixin/core/world/MixinWorld.java

```diff
@@ -95,6 +111,38 @@ public void onConstructed(ISaveHandler saveHandlerIn, WorldInfo info, WorldProvi
         }
     }
 
+    /**
+     * @author Simon816
+     *
+     * @see org.spongepowered.common.mixin.core.world.MixinWorld#spawnEntityInWorld
+     * Only difference is that this includes forge's changes.
+     */
+    @Overwrite
+    public boolean spawnEntityInWorld(net.minecraft.entity.Entity entityIn) {
+        if (!this.isRemote && (entityIn == null || (entityIn instanceof net.minecraft.entity.item.EntityItem && this.restoringBlockSnapshots))) {
+            return false;
+        }
+        int cx = MathHelper.floor_double(entityIn.posX / 16.0D);
+        int cz = MathHelper.floor_double(entityIn.posZ / 16.0D);
+        boolean force = entityIn.forceSpawn || (entityIn instanceof EntityPlayer && !(entityIn instanceof HumanEntity));
```

Comment:
When this was made there was no such thing as capturing locals. I'll see if I can get that to work